realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -203,7 +202,12 @@ private void postError(final ObjectServerError error) {
                     handler.post(new Runnable() {
                         @Override
                         public void run() {
-                            callback.onError(error);
+                            try {
```

Comment:
I this correct? `loginAsync` will called mostly from UI thread, and the this `Runnable` is posted to UI thread, it should be totally valid to throw an exception from here.